### PR TITLE
Add step description

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

### Changes

Add detailed step description about:

how key-based caching works
how the key templating works
how to define multiple keys

image

Expanded cache key input:
image

### CODE REVIEW

File: step.yml [hunk 1]
```diff
@@ -1,6 +1,55 @@
...
+
+  Caches can become outdated across builds when something changes in the project (for example, a dependency gets upgraded to a new version). In this case, a new (unique) cache key is needed to save the new cache contents. This is possible if the cache key is dynamic and changes based on the project state (for example, a checksum of the dependency lockfile is part of the cache key). If you use the same dynamic cache key when restoring the cache, the Step will download the most relevant cache archive available.
+
+  Key-based caching is platform-agnostic and can be used to cache anything by carefully selecting the cache key and the files/folders to include in the cache.
+
+  #### Templates
+
+  The Step requires a string key to use when uploading a cache archive. In order to always download the most relevant cache archive for each build, the cache key input can contain template elements. The **Restore cache Step** evaluates the key template at runtime and the final key value can change based on the build environment or files in the repo. Similarly, the **Save cache** step also uses templates to compute a unique cache key when uploading a cache archive.
```
Comment (daniCsorba) [inline step.yml:16]:
```suggestion
  The Step requires a string key to use when uploading a cache archive. In order to always download the most relevant cache archive for each build, the cache key input can contain template elements. The **Restore cache Step** evaluates the key template at runtime and the final key value can change based on the build environment or files in the repo. Similarly, the **Save cache** Step also uses templates to compute a unique cache key when uploading a cache archive.
```

File: docs/examples.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+steps:
+- save-cache@1:
+    inputs:
+    - key: '{{ .OS }}-{{ .Arch }}-npm-cache-{{ checksum "package-lock.json" }}'
+    - path: node_modules
+```
+
+#### Multiple independent caches
+
+You can add multiple instances of this step to a workflow:
```
Comment (daniCsorba) [inline docs/examples.md:36]:
```suggestion
You can add multiple instances of this Step to a Workflow:
```

File: docs/examples.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+- save-cache@1:
+    run_if: ".IsCI | and (not .IsPR)" # Condition that is false in PR builds
+    inputs:
+    - key: node-modules-{{ checksum "package-lock.json" }}
+    - paths: node_modules
+```
+
+#### Separate caches for each OS and architecture
+
+Cache is not guaranteed to work across different Bitrise Stacks (different OS or same OS but different CPU architecture). If a workflow runs on different stacks, it's a good idea to include the OS and architecture in the cache key:
```
Comment (daniCsorba) [inline docs/examples.md:24]:
```suggestion
Cache is not guaranteed to work across different Bitrise Stacks (different OS or same OS but different CPU architecture). If a Workflow runs on different stacks, it's a good idea to include the OS and architecture in the **Cache key** input:
```

**Review by daniCsorba — CHANGES_REQUESTED** [review]
Submitting comments from technical writing perspective.

**Review by tothszabi — APPROVED** [review]

**Pull request merged**